capsules/src/ninedof.rs

```diff
@@ -51,21 +53,45 @@ impl Default for App {
     }
 }
 
+pub struct NineDofNode<'a, P> {
+    driver: P,
+    next: ListLink<'a, NineDofNode<'a, P>>,
+}
+
+impl<P> NineDofNode<'a, P> {
+    pub fn new(driver: P) -> Self {
+        NineDofNode {
+            driver: driver,
+            next: ListLink::empty(),
+        }
+    }
+}
+
+impl<P> ListNode<'a, NineDofNode<'a, P>> for NineDofNode<'a, P> {
+    fn next(&'a self) -> &'a ListLink<'a, NineDofNode<'a, P>> {
+        &self.next
+    }
+}
+
 pub struct NineDof<'a> {
-    driver: &'a dyn hil::sensors::NineDof,
+    drivers: List<'a, NineDofNode<'a, &'a dyn hil::sensors::NineDof>>,
```

Comment:
What about using something like LED as a template and using a slice instead? This would reduce complexity, and I don't think we need the flexibility of the List approach.

That would also simplify the component interface, and remove the possibility of forgetting to create a `NineDofDriverComponent`.

Reply:
Thank you for the feedback, make much more sens like this.